### PR TITLE
Small doxygen fixes

### PR DESCRIPTION
**Pull request by lurch**
Small doxygen fixups (gets rid of a few errors / warnings from doxygen's output)

### CODE REVIEW

File: src/rp2_common/hardware_gpio/include/hardware/gpio.h [hunk 1]
```diff
@@ -476,9 +476,8 @@ static inline uint32_t gpio_get_irq_event_mask(uint gpio) {
  * the opposite level).
  *
  * \param gpio GPIO number
- * \param events Bitmask of events to clear. See \ref gpio_set_irq_enabled for details.
  *
- * \note For callbacks set with \ref gpio_set_irq_enabled_with_callback, or \ref gpio_set_irq_callback,this function is called automatically.
+ * \note For callbacks set with \ref gpio_set_irq_enabled_with_callback, or \ref gpio_set_irq_callback, this function is called automatically.
  * \param event_mask Bitmask of events to clear. See \ref gpio_irq_level for details.
```
Comment (peterharperuk) [inline src/rp2_common/hardware_gpio/include/hardware/gpio.h:481]: Should the final \param be higher up? Before the \note?

File: src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h [hunk 2]
```diff
@@ -194,13 +194,13 @@ int cyw43_arch_wifi_connect_blocking(const char *ssid, const char *pw, uint32_t
...
  * \param ssid the network name to connect to
- * \param password the network password or NULL if there is no password required
+ * \param pw the network password or NULL if there is no password required
  * \param auth the authorization type to use when the password is enabled. Values are \ref CYW43_AUTH_WPA_TKIP_PSK,
  *             \ref CYW43_AUTH_WPA2_AES_PSK, or \ref CYW43_AUTH_WPA2_MIXED_PSK (see \ref CYW43_AUTH_)
  *
  * \return 0 if the initialization is successful, an error code otherwise \see pico_error_codes
  */
-int cyw43_arch_wifi_connect_timeout_ms(const char *ssid, const char *pw, uint32_t auth, uint32_t timeout);
+int cyw43_arch_wifi_connect_timeout_ms(const char *ssid, const char *pw, uint32_t auth, uint32_t timeout_ms);
```
Comment (peterharperuk) [inline src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h:203]: timeout_ms is missing from the documentation?

File: src/rp2_common/hardware_gpio/include/hardware/gpio.h (hunk 1 above)
Comment (lurch) [reply src/rp2_common/hardware_gpio/include/hardware/gpio.h:481]: I wasn't sure - Doxygen doesn't complain and this minimises the diff :shrug: 
Happy to move it up a few lines if you think it'd look better though.

File: src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h (hunk 2 above)
Comment (lurch) [reply src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h:203]: Yup - I don't know the code well enough to add a description, but I can add the missing `\param timeout_ms` if you suggest something suitable.
(the actual change I'm making here is to rename the parameter in the `.h` to match the parameter-name in the `.c`)

**Review by kilograham — APPROVED** [review]

**Pull request merged**